src/components/CodeAssignmentModal/index.jsx

```diff
@@ -299,16 +296,10 @@ export class BaseCodeAssignmentModal extends React.Component {
         enterprise_customer: enterpriseUuid,
       };
     }
-
+    const assignmentEmails = isBulkAssign ? validEmails : [formData['email-address']];
```

Comment:
can we reuse the if statement we already have?
```
const assignmentEmails;
if (isBulkAssign) {
      assignmentEmails = validEmails;

      // Only includes `codes` in `options` if not all codes are selected.
      if (!hasAllCodesSelected) {
        options.codes = selectedCodes.map(selectedCode => selectedCode.code);
      }
    } else {
      assignmentEmails = [formData['email-address']];
      options.codes = [code.code];
    }

```

Reply:
In that case, code will be moved in else block  if `isBulkAssign`  `hasAllCodesSelected` is true which is not desired

`if (isBulkAssign && !hasAllCodesSelected) {`